Consistency fixes for i.MX8M SOC

dbaluta opened the pull request:
This PR contains 2 patches fixing i.MX8MP SOC
* first use mimx8ml8 instead of imx8mp for SOC name
* use "_" instead of "." as board qualifier separator when creating Kconfig name

dbaluta committed:
soc: nxp: imx8: Fix Kconfig.defconfig.* filenames

We need to use correct SOC name when naming the Kconfig.defconfig*
files.

So we need the following changes:
	- mimx8mp -> mimx8ml8
	- mimx8mm -> imx8mm6
	- mimx8mn -> mimx8mn6
	- mimx8mq -> mimx8mq6

Then we also need to take care of qualifiers name. Standard notation
uses "_" instead of "."

e.g : Kconfig.defconfig.mimx8mp.a53 -> Kconfig.defconfig.mimx8ml8_a53

Signed-off-by: Daniel Baluta <daniel.baluta@nxp.com>

dbaluta commented:
> @dbaluta, I found we also need to unify the file name for imx8mm and imx8mq in this directory, would you please also update these file names or I will update late? thanks.

Yes, I also noticed that and my intention was to fix that in a separate PR to make this easier to review and merge faster :).

JiafeiPan commented:
> > @dbaluta, I found we also need to unify the file name for imx8mm and imx8mq in this directory, would you please also update these file names or I will update late? thanks.
> 
> Yes, I also noticed that. I will take care of it.
> 
> Do you think should I add that changes in this PR or create another one?

I think we can cover all these updates in this PR, thanks.

dbaluta commented:
@JiafeiPan  fixed.

Changes since v1: 
- fixed Kconfig filenames for imx8mm, imx8mn, imx8mq as per @JiafeiPan  feedback
- squashed everything under 1 single patch as the changes are small and can be easily reviewed
